cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/AdaptTemplatesCommand.java

```diff
@@ -0,0 +1,239 @@
+                        fileSystem.getPath(templatesFolderPath + File.separator + entry.getName());
+                } else if (entry.getName().contains("com/")) {
+                    saveForFileCreationPath = fileSystem.getPath(templatesFolderPath + File.separator + "src"
+                        + File.separator + "main" + File.separator + "java" + File.separator + entry.getName());
+                }
+                createFile(file, entry, saveForFileCreationPath);
+            }
+        } catch (IOException e) {
+            throw new IOException("An exception occurred while unpacking sources from Jar file to templates folder");
+        }
+
+        // unpack classes to target directory
+        try (ZipFile file = new ZipFile(classesJarPath)) {
+            Enumeration<? extends ZipEntry> entries = file.entries();
+            Path sourcesClassPath = fileSystem.getPath(templatesFolderPath + File.separator + "target");
+            if (Files.notExists(sourcesClassPath)) {
+                Files.createDirectory(sourcesClassPath);
+            }
+            while (entries.hasMoreElements()) {
+                ZipEntry entry = entries.nextElement();
+                if (entry.getName().contains("com/")) {
+                    Path saveForFileCreationPath = fileSystem.getPath(sourcesClassPath + File.separator + "src"
+                        + File.separator + "main" + File.separator + "java" + File.separator + entry.getName());
+                    createFile(file, entry, saveForFileCreationPath);
+                }
+            }
+        } catch (IOException e) {
+            throw new IOException("An exception occurred while unpacking classes from Jar files to templates folder");
+        }
+    }
+
+    /**
+     * Creates a file or directory and cleans up current file if it exists already
+     * @param file
+     *            ZipFile to access
+     * @param entry
+     *            ZipEntry to get input stream from
+     * @param saveForFileCreationPath
+     *            Path to save file at
+     * @throws IOException
+     *             if file could not be created
+     */
+    private static void createFile(ZipFile file, ZipEntry entry, Path saveForFileCreationPath) throws IOException {
+        if (entry.isDirectory()) {
+            Files.createDirectories(saveForFileCreationPath);
+        } else {
+            Files.deleteIfExists(saveForFileCreationPath);
+            Files.createFile(saveForFileCreationPath);
+            try (InputStream is = file.getInputStream(entry);
+                BufferedInputStream bis = new BufferedInputStream(is);
+                FileOutputStream fileOutput = new FileOutputStream(saveForFileCreationPath.toString());) {
+
+                while (bis.available() > 0) {
+                    fileOutput.write(bis.read());
+                }
+            } catch (IOException e) {
+                throw new RuntimeIoException(e, IoMode.WRITE);
+            }
+        }
```

Comment:
Please check for apache-commons functions to write files, do not to this on your own.